botocore/client.py

```diff
@@ -218,6 +221,28 @@ def _get_client_args(self, service_model, region_name, is_secure,
             config_kwargs.update(
                 connect_timeout=client_config.connect_timeout,
                 read_timeout=client_config.read_timeout)
+
+        # Determine if the user specified the use of virtual host addressing
+        # style for s3.
+        s3_addressing_style = 'default'
```

Comment:
Can you move this out to a separate method?  This method is getting pretty long.